member.php

```diff
@@ -125,7 +125,15 @@
 
 	if($mybb->settings['regtype'] == "randompass")
 	{
-		$mybb->input['password'] = random_str();
+		$mybb->settings['requirecomplexpasswords'] = 0;
+
+		$password_length = (int)$mybb->settings['minpasswordlength'];
+		if($password_length < 8)
+		{
+			$password_length = min(8, (int)$mybb->settings['maxpasswordlength']);
+		}
+
+		$mybb->input['password'] = random_str($password_length, true);

```

Comment:
The 2nd parameter could be `false`/`true` based on `$mybb->settings['requirecomplexpasswords']` value (ofc after removing the `0` assignment I commented above).